pkg/generated/openapi/openapi_test.go

```diff
@@ -24,12 +24,15 @@ import (
 	"github.com/go-openapi/spec"
 
 	"k8s.io/apimachinery/pkg/util/diff"
+	"k8s.io/kube-openapi/pkg/handler"
 )
 
 func TestOpenAPIRoundtrip(t *testing.T) {
 	dummyRef := func(name string) spec.Ref { return spec.MustCreateRef("#/definitions/dummy") }
 	for name, value := range GetOpenAPIDefinitions(dummyRef) {
 		t.Run(name, func(t *testing.T) {
+			// TODO(kubernetes/gengo#193): We currently round-trip ints to floats.
```

Comment:
what is the impact of this int -> float issue here?

Reply:
We have a benchmark here https://github.com/kubernetes/gengo/pull/192#discussion_r520955114.

The Protocol string field is the only field that we are looking to apply this defaulting to for 1.20 and we already have a follow up task to use direct comparisons for primitive types.

Reply:
None.

* Round tripping here is only done for the test.
* CRD defaults are not problematic because they don't go through the generated default code.
* `// +default=` directives are not problematic because there is only a single one in 1.20 and we know it's fine.
* Before we open the floodgates to let people set more defaults, @Jefftree will update the default generation code to perform a typecheck and barf if the author is trying to stuff an int into a float in a way that loses precision, along with all other problems that can be caught by a typecheck. But this doesn't matter for 1.20, since we won't be adding more of these in 1.20.

Reply:
> * `// +default=` directives are not problematic because there is only a single one in 1.20 and we know it's fine.

A single one in-tree... but don't we have to handle data coming from aggregated servers as well that could start generating default-containing openapi?

This test was added to detect things that would make the merging controller falsely detect differences and churn re-merging "updated" specs - https://github.com/kubernetes/kubernetes/pull/71223#discussion_r254046361

I wanted to make sure we're not reintroducing that possibility. I still can't tell if we are.

Reply:
> This test was added to detect things that would make the merging controller falsely detect differences and churn re-merging "updated" specs 

The merging controller strips the default like this test does. Isn't it testing the right thing then?

Reply:
> but don't we have to handle data coming from aggregated servers as well that could start generating default-containing openapi?

They wouldn't publish these.

Reply:
> don't we have to handle data coming from aggregated servers as well that could start generating default-containing openapi?

No. They're on their own for running SSA, and we strip defaults before publishing.

Reply:
> The merging controller strips the default like this test does. Isn't it testing the right thing then?

If that changes, will we remember to change here in the same way? Should we use a "prepare for publishing" func consistently between the two? I just want to make sure this test will continue catching the problems that prompted its addition.

Reply:
We do want to fix this ASAP, and we absolutely want to fix it before we consider publishing the defaults.